Various improvements to the Hot Reload docs (#7396)
MicrosoftDocs/visualstudio-docs: docs/debugger/hot-reload.md

The following table shows which application types support .NET Hot Reload with the debugger attached (F5) and without the debugger attached (Ctrl+F5), and whether .NET 6 is required for minimum support (that is, F5). .NET 6 is always required for Ctrl+F5 support. Also shown is the minimum version of Visual Studio that supports the feature.

|Application type|.NET 6 required|F5|Ctrl+F5|
|-|-|-|-|
|ASP.NET code-behind|No|16.11|17.0|
|ASP.NET Razor (Blazor Server and ASP.NET Core)|Yes|17.0|17.0|
|ASP.NET Razor (Blazor WASM)|Yes|17.1|17.0|
|WPF|No|16.11|17.0|
|WinUI3|No|16.11|--|
|WinForms|No|16.11|17.0|
|Console|No|16.11|17.0|
|.NET MAUI (WinUI 3)|Yes|17.1|--|
|.NET MAUI (Android)|Yes|17.1|--|
|.NET MAUI (iOS)|Yes|17.1|--|
|.NET MAUI Blazor Hybrid (WinUI 3)|Yes|17.1|--|
|.NET MAUI Blazor Hybrid (Android)|Yes|17.1|--|

> |.NET MAUI Blazor Hybrid (iOS)|Yes|17.1|--|

Reviewer: The table looks OK to me. It's always good to do this kind of thing on the private repo, since we have build validation there, but we can see what happens.

When using Visual Studio 2022 and starting the app with the debugger, Hot Reload works with most app frameworks, including typical app types such as Console, Windows Forms (WinForms), WPF, UWP, WinUI 3 (see note), and most types of ASP.NET web projects (for code-behind edits) including ASP.NET MVC, Web API, and even older Web Forms projects. These are examples. Anywhere you have .NET and you’re using the Visual Studio managed debugger, you should get basic Hot Reload support. This fact means that even projects such as Azure Functions work great in this scenario.

> [!NOTE]

> > WinUI 3 by default uses mixed mode debugging, which does not support Hot Reload. You can modify this in project settings by enabling the Managed Debugger, which enables Hot Reload to work properly. To enable this in your project modify launchSettings.json and add `"nativeDebugging": false` under commandName property

Reviewer: ```suggestion
> WinUI 3 by default uses mixed mode debugging, which does not support Hot Reload. You can modify this in project settings by enabling the Managed Debugger, which enables Hot Reload to work properly. To enable this in your project modify launchSettings.json and add `"nativeDebugging": false` after the `commandName` property
```

Reviewer: I'd say "after" instead of under to try to avoid implying that it's nested. As far as the example, maybe that would be helpful. If commandName is always present in all WinUI scenarios, it's probably OK and people might not need it.

* UWP apps are not supported for Hot Reload without the debugger. This is by design, and there are no current plans to improve this.
* Xamarin.Forms apps targeting iOS and Android don’t support .NET Hot Reload (regardless of whether you are starting your app with or without the debugger), but will continue to support XAML Hot Reload.

> * .NET MAUI apps are not supported without the debugger.

Reviewer: ```suggestion
* .NET MAUI apps are supported only with the debugger.
```

Reviewer: Avoiding the double negative not / without

![Screenshot of settings for .NET Hot Reload](../debugger/media/vs-2022/dotnet-hot-reload-settings.png)

You can also control if .NET Hot Reload is available at the project level by modifying your .NET 6 projects launchSetting.json and setting `hotReloadEnabled` to `false`

Example:

> ~~~XAML

Reviewer: ```suggestion
```XAML
```

{
  "profiles": {
    "Console": {
      "commandName": "Project",
      "hotReloadEnabled": false
    }
  }
}

> ~~~

Reviewer: ```suggestion
```
```

Reviewer: Weird, I tried to replace ~~~ with ```, but it shows up strangely when rendered here. Can you make sure we use ```?
